src/threading/MsgThread.cc

```diff
@@ -227,6 +225,83 @@ MsgThread::MsgThread() : BasicThread(), queue_in(this, nullptr), queue_out(nullp
 
     io_source = new detail::IOSource(this);
```

Comment:
Would keep this close to the `io_mgr->Register()` call (or move the Register() call up).